<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [7]:
# Membuat dataframes dari dictionary
import pandas as pd
data = {
    "Nama klub": ["persib", "persebaya", "persija", "arema"],
    "tahun lahir": [1919, 1927, 1928, 1987],
    "Kota": ["Bandung", "Surabaya", "Jakarta", "Malang"]
}


df = pd.DataFrame(data)


df

Nama klub  tahun lahir      Kota
0     persib         1919   Bandung
1  persebaya         1927  Surabaya
2    persija         1928   Jakarta
3      arema         1987    Malang

<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [17]:
# Membaca data dan memasukannya kedalam suatu variabel
data=pd.read_csv('diamonds-1.csv')

In [18]:
data.head(5)

Unnamed: 0  carat      cut color clarity  depth  table  price     x     y  \
0           1   0.23    Ideal     E     SI2   61.5   55.0    326  3.95  3.98   
1           2   0.21  Premium     E     SI1   59.8   61.0    326  3.89  3.84   
2           3   0.23     Good     E     VS1   56.9   65.0    327  4.05  4.07   
3           4   0.29  Premium     I     VS2   62.4   58.0    334  4.20  4.23   
4           5   0.31     Good     J     SI2   63.3   58.0    335  4.34  4.35   

      z  
0  2.43  
1  2.31  
2  2.31  
3  2.63  
4  2.75

In [19]:
data.tail(5)

Unnamed: 0  carat        cut color clarity  depth  table  price     x  \
53935       53936   0.72      Ideal     D     SI1   60.8   57.0   2757  5.75   
53936       53937   0.72       Good     D     SI1   63.1   55.0   2757  5.69   
53937       53938   0.70  Very Good     D     SI1   62.8   60.0   2757  5.66   
53938       53939   0.86    Premium     H     SI2   61.0   58.0   2757  6.15   
53939       53940   0.75      Ideal     D     SI2   62.2   55.0   2757  5.83   

          y     z  
53935  5.76  3.50  
53936  5.75  3.61  
53937  5.68  3.56  
53938  6.12  3.74  
53939  5.87  3.64

Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawab:



*   Series, list, dan dictionary adalah struktur data yang berbeda. List adalah kumpulan data yang terurut dan dapat diubah, sedangkan dictionary adalah kumpulan pasangan kunci-nilai. Series adalah struktur data satu dimensi yang merupakan bagian dari library Pandas, yang memungkinkan manipulasi dan analisis data yang efisien. Series memiliki indeks yang dapat disesuaikan dan mendukung operasi aritmatika dan statistik. Berbeda dengan list, series memiliki kemampuan untuk melakukan operasi secara vektoris dan mendukung data waktu dan tanggal.

*   DataFrame dan Series adalah dua struktur data utama dalam library Pandas. Series adalah struktur data satu dimensi yang menyerupai kolom tunggal dalam tabel, sedangkan DataFrame adalah struktur data dua dimensi yang menyerupai tabel dengan baris dan kolom. DataFrame terdiri dari kumpulan Series yang terkait dengan indeks yang sama. DataFrame mendukung operasi seperti penggabungan, pengurutan, dan pengelompokan data, sedangkan Series lebih cocok untuk operasi statistik dan analisis data tunggal.



In [25]:
# code here
import pandas as pd

# Membuat data Series
data_series = pd.Series([41, 38, 37, 32, 31], index=["persib", "persebaya", "persija", "dewa united", "persik"], name="klasemen liga 1")

# Menampilkan Series
data_series

persib         41
persebaya      38
persija        37
dewa united    32
persik         31
Name: klasemen liga 1, dtype: int64

<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [22]:
# code here
df = pd.DataFrame(data)

first_rows = df.head(3)
last_rows = df.tail(3)

first_price = df.loc[0, "price"]
last_color = df.loc[df.index[-1], "color"]

first_price, last_color

(326, 'D')

<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Jawab:

Deskriptif statistik adalah metode analisis data yang berfokus pada pengumpulan, pengolahan dan penyajian data secara numerik dan grafis untuk memahami karakteristik data. Ini meliputi ukuran pusat (rata-rata, median, modus), ukuran variasi (rentang, variansi, simpangan baku), dan ukuran keragaman (koefisien variasi, indeks keragaman). Data disajikan dalam bentuk tabel, grafik (batang, lingkaran, garis), diagram, histogram dan grafik kotak-sayap. Analisis ini membantu mengidentifikasi pola, tren dan karakteristik data, sehingga memudahkan pengambilan keputusan dalam berbagai bidang seperti bisnis, kesehatan, pendidikan dan penelitian.

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [27]:
# code here

df = pd.DataFrame(data)


stats = df.describe()
mean_price = stats.loc["mean", "price"]
std_depth = stats.loc["std", "depth"]
max_carat = stats.loc["max", "carat"]

mean_price, std_depth, max_carat


(3932.799721913237, 1.4326213188337733, 5.01)

Jawab:

berikut adalah hasil analisis deskriptif statistik berdasarkan data pada dataset diamonds-1:
*   Nilai rata-rata dari kolom price: 3932.799721913237
*   Nilai standar deviasi dari kolom depth: 1.4326213188337733
*   Nilai maksimum dari kolom carat: 5.01




